SpellupRecast/Hadar_Spellups.xml

```diff
@@ -1037,17 +1037,17 @@ function cast_spells()
 			if contains(Spellups["EX"]["noAuto"],i) == false then
 
 				if i == 518 then
-                              if Spellups["NA"][552] == 1 then --if not affected by gaias focus
+					if Spellups["NA"][552] == 1 then --if not affected by gaias focus
 						cast_this(552,tonumber(Spellups["AS"]["list"][552]["type"]))
 						cast_this(518,tonumber(Spellups["AS"]["list"][518]["type"]))
 						hadarprint("Report NA: "..Spellups["NA"][552],"debug")
-                              	hadarprint("Sending "..Spellups["AS"]["list"][552]["name"].." to cast","debug")                                     
-                              else
+						hadarprint("Sending "..Spellups["AS"]["list"][552]["name"].." to cast","debug")
+					elseif Spellups["NA"][552] == 0 then -- player is affected by gaias focus
 						cast_this(518,tonumber(Spellups["AS"]["list"][518]["type"]))
 						hadarprint("Gaias focus reported ON, not casting this time","debug") --why reported on '0' when it was off?
 						hadarprint("Report NA: "..Spellups["NA"][552],"debug")
```

Comment:
The error would occur on line 1048 when concatenating `Spellups["NA"][552]` because `552` is not an entry in `"NA"`

I added the following on line 1038:
```
                require "tprint"
				print("DBG: i == " .. i)
				print("Will tprint  Spellups['NA']:")
				tprint(Spellups["NA"])
				print("Done tprinting...")
```

And, when 518 finished, got the following log:
```
DBG: i == 518
Will tprint  Spellups['NA']:
67=0
235=0
49=0
423=0
422=0
212=0
48=0
424=0
431=0
430=0
429=0
32=1
252=1
253=1
254=1
255=1
157=1
317=1
511=0
249=1
155=0
250=1
72=0
91=1
152=1
321=0
335=1
171=0
333=0
259=0
606=0
140=0
5=0
226=1
68=0
4=0
160=0
58=0
139=0
71=1
51=0
440=0
518=1
34=1
31=1
57=0
Done tprinting...
```

552 is not a key. I'm guessing it is because I never cast it?

I only found `Spellups["NA"][X]` being set on `affon` and `affoff`, so we probably never set 552 there and thus get the nil when trying to concat

Reply:
PR #43 should fix this